### PR TITLE
added challenge completion status

### PR DESCRIPTION
**Pull request by Vladmidir**
# Summary
Added 'completed' attribute to each Challenge page and ChallengeCard. Implemented hiding completed challenges.

# Screenshots
image
image
image

### CODE REVIEW

File: client/src/pages/Challenge.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ const Challenge = () => {
         // console.log(details);
         return;
       })
-      .catch((err) => {
+      .catch((err: Error) => {
```
Comment (alexapostolu) [inline client/src/pages/Challenge.tsx:36]: Is adding the Error type here necessary?

File: client/src/pages/Challenge.tsx [hunk 2]
```diff
@@ -11,7 +11,7 @@ import InstructionsPopup from '../components/InstructionsPopup';
 const Challenge = () => {
     //TODO: Fetch this value from the server. 
     // We can query the db for solutions with the current user id and the challenge id
-    const [completed, setCompleted] = useState(false);
+    // const [completed, setCompleted] = useState(false);
```
Comment (arthurhohinng) [inline client/src/pages/Challenge.tsx:14]: whys this commented

Comment (arthurhohinng) [reply client/src/pages/Challenge.tsx:14]: can we remove?

**Review by arthurhohinng — APPROVED** [review]

File: client/src/pages/Challenge.tsx (hunk 1 above)
Comment (Vladmidir) [reply client/src/pages/Challenge.tsx:36]: typescript will be mad at you if you dont

File: client/src/pages/Challenge.tsx (hunk 2 above)
Comment (Vladmidir) [reply client/src/pages/Challenge.tsx:14]: yes

**Pull request merged**